server/blob/blob-joining/src/main/java/org/apache/james/blob/joining/JoiningBlobStore.java

```diff
@@ -146,13 +91,43 @@ public InputStream read(BlobId blobId) {
         }
     }
 
-    @VisibleForTesting
-    BlobStore getPrimaryBlobStore() {
-        return primaryBlobStore;
+    private Supplier<CompletableFuture<BlobId>> saveToPrimaryFallbackIfFails(
+        Supplier<CompletableFuture<BlobId>> primarySavingOperation,
+        Supplier<CompletableFuture<BlobId>> fallbackSavingOperation) {
```

Comment:
Can't the signature be:

```
 private CompletableFuture<BlobId> saveToPrimaryFallbackIfFails(
        CompletableFuture<BlobId> primarySavingOperation,
        Supplier<CompletableFuture<BlobId>> fallbackSavingOperation)
```

This would avoid useless (?) wrapping/unwrapping....